modules/python/src2/cv2.cpp

```diff
@@ -206,6 +206,23 @@ bool pyopencv_to(PyObject* obj, T& p, const char* name = "<unknown>");
 template<typename T> static
 PyObject* pyopencv_from(const T& src);
 
+template <typename T> static
+bool pyopencv_to_ptr(PyObject *o, T* p, const char *name = "<unknown>")
+{
+    if (!o || o == Py_None)
+        return true;
+    p = new T();
```

Comment:
@mshabunin You are right! memory leakage is now avoided using the smart pointers.
I have revised the code accordingly, please confirm...

However, IMHO, the fact that the new Python wrapper wraps all functionalities properly and is passing all current tests is enough for a test.

Furthermore, the code generated from the new wrapper would solve issues that we were not able to solve previously.

For example, `cv::EMD`'s `lowerBound` parameter wraps was:
```C++
    float lowerBound = 0;
```
which is obviously incorrect, and caused troubles for years (such as #5565) and ugly workarounds (such as #8496) were recently utilized to solve that.

Now it has become:
```C++
    Ptr<float> lowerBound;
```
Therefore, with the new wrapper, we should be able to handle these cases in a better way.

However, I cannot provide direct tests for this PR alone, and the tests I provided in #8550 are still conflicting with Java wrapper, but they prove the new Python wrapper is totally sound.